### PR TITLE
Add support for grids

### PR DESCRIPTION
**Pull request by sondove**
This adds very basic support for grids to the novem python package and cli

### CODE REVIEW

File: tests/test_cli_grids.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
...
+
+    # try to list all grids
+    params = ["-g", "-l"]
+
+    # launch cli with params
+    sys.argv = ["novem"] + params
+
+    # run cli
+    run_cli()
+    out, err = capsys.readouterr()
```
Comment (bjornars) [inline tests/test_cli_grids.py:188]: I put a nice little `cli` fixture in `conftest.py`, see usage in https://github.com/novem-code/novem-python/blob/main/tests/test_cli_config.py#L61

File: novem/vis/grid.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        # first let our super do it's thing
+        super()._parse_kwargs(**kwargs)
+
+        # get a list of valid properties
+        # exclude mapping and layout as they needs to be run last
+        props = [x for x in dir(self) if x[0] != "_" and x not in ["layout", "mapping", "read", "delete", "write"]]
+
+        do_layout = False
+        do_mapping = False
+        for k, v in kwargs.items():
```
Comment (bjornars) [inline novem/vis/grid.py:65]: we're trying to get away from kwargs, not add more!

Comment (sondove) [reply novem/vis/grid.py:65]: This is just a convenient way to ensure that all `@property` setters are also available as `kwargs` on the `Class` init.

The reason for this is to ensure that both interactive use of novem like this:
```python
plt = Plot('test_plot')
plt.type = "gbar"
plt.title = "Simple test bar"
plt.data = df
plt.x
```

can also be done inline like this
```python
df.pipe(Plot("test_plot", type="gbar", title="Simple test bar")).x
```

Perhaps there is a better way of doing this, but with the current version adding a `@property` also introduces automatic support for `kwarg` parsing. So it's somewhat DRY.

This was a copy of existing code, so it is at least consistent. I encourage a possible rewrite and better documentation of this behaviour, but I'm not going to do it for this PR

File: tests/test_cli_grids.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+        return
+
+    requests_mock.register_uri(
+        "post",
+        f"{api_root}vis/grids/{grid_name}/layout",
+        text=partial(set_caption, content),
+    )
+
+    # try to delete a no existant grid
+    params = ["-g", grid_name]
```
Comment (bjornars) [inline tests/test_cli_grids.py:335]: lies

**Review by bjornars — APPROVED** [review]
Just merge it, I'll fix. Fix fix.

**Pull request merged**